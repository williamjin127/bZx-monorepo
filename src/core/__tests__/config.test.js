/* globals test, expect, describe */
import Web3 from "web3";
import B0xJS from "../../core";

describe("config", () => {
  test("should use default config when config arg not provided", () => {
    const networkUrl = "http://localhost:8545";
    const provider = new Web3.providers.HttpProvider(networkUrl);
    const b0xJS = new B0xJS(provider);

    expect(b0xJS.addresses).toMatchSnapshot();
  });

  test("should use provided config", () => {
    const networkUrl = "http://localhost:8545";
    const provider = new Web3.providers.HttpProvider(networkUrl);
    const config = {
      addresses: {
        B0x: "thisisatest",
        B0xOracle: "thisisatest",
        B0xTo0x: "thisisatest",
        B0xToken: "thisisatest",
        B0xVault: "thisisatest",
        EIP20: "thisisatest",
        OracleRegistry: "thisisatest",
        TestToken0: "thisisatest",
        TestToken1: "thisisatest",
        TestToken2: "thisisatest",
        TestToken3: "thisisatest",
        TestToken4: "thisisatest",
        TestToken5: "thisisatest",
        TestToken6: "thisisatest",
        TestToken7: "thisisatest",
        TestToken8: "thisisatest",
        TestToken9: "thisisatest",
        TokenRegistry: "thisisatest"
      }
    };

    const b0xJS = new B0xJS(provider, config);

    expect(b0xJS.addresses).toMatchSnapshot();
  });
});
