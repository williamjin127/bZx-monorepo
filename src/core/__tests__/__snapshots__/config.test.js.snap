// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should use default config when config arg not provided 1`] = `
Object {
  "B0x": "0x8726c7414ac023d23348326b47af3205185fd035",
  "B0xOracle": "0x371b13d97f4bf77d724e78c16b7dc74099f40e84",
  "B0xTo0x": "0x32eecaf51dfea9618e9bc94e9fbfddb1bbdcba15",
  "B0xToken": "0x04b5dadd2c0d6a261bfafbc964e0cac48585def3",
  "B0xVault": "0xb7c9b454221e26880eb9c3101b3295ca7d8279ef",
  "EIP20": "",
  "OracleInterface": "",
  "OracleRegistry": "0x6000eca38b8b5bba64986182fe2a69c57f6b5414",
  "TestNetFaucet": "0xf96b018e8de3a229dbaced8439df9e3034e263c1",
  "TestToken0": "0x4586649629f699f9a4b61d0e962dc3c9025fe488",
  "TestToken1": "0x5d3ad3561a1235273cbcb4e82fce63a0073d19be",
  "TestToken2": "0xb48e1b16829c7f5bd62b76cb878a6bb1c4625d7a",
  "TestToken3": "0xc4cc602a7345518d0b7a84049d4bc8575ebf3398",
  "TestToken4": "0xe704967449b57b2382b7fa482718748c13c63190",
  "TestToken5": "0xa4b3e1659c473623287b2cc13b194705cd792525",
  "TestToken6": "0x2c530e4ecc573f11bd72cf5fdf580d134d25f15f",
  "TestToken7": "0x72d5a2213bfe46df9fbda08e22f536ac6ca8907e",
  "TestToken8": "0x2ebb94cc79d7d0f1195300aaf191d118f53292a8",
  "TestToken9": "0x5315e44798395d4a952530d131249fe00f554565",
  "TokenRegistry": "0xc4df27466183c0fe2a5924d6ea56e334deff146a",
}
`;

exports[`config should use provided config 1`] = `
Object {
  "B0x": "thisisatest",
  "B0xOracle": "thisisatest",
  "B0xTo0x": "thisisatest",
  "B0xToken": "thisisatest",
  "B0xVault": "thisisatest",
  "EIP20": "thisisatest",
  "OracleInterface": "thisisatest",
  "OracleRegistry": "thisisatest",
  "TestToken0": "thisisatest",
  "TestToken1": "thisisatest",
  "TestToken2": "thisisatest",
  "TestToken3": "thisisatest",
  "TestToken4": "thisisatest",
  "TestToken5": "thisisatest",
  "TestToken6": "thisisatest",
  "TestToken7": "thisisatest",
  "TestToken8": "thisisatest",
  "TestToken9": "thisisatest",
  "TokenRegistry": "thisisatest",
}
`;
